analyzer/codechecker_analyzer/analyzers/cppcheck/analyzer.py

```diff
@@ -27,6 +28,18 @@
 LOG = get_logger('analyzer.cppcheck')
 
 
+def parse_version(cppcheck_output):
+    """
+    Parse cppcheck version output and return the version number.
+    """
+    version_re = re.compile(r'^Cppcheck (?P<version>[\d\.]+)')
+    match = version_re.match(cppcheck_output)
+    if match:
+        return float(match.group('version'))
```

Comment:
Using float to represent version is not the most robut solution. I would parse major, minor, patch separately, or as a tuple. This has a benefit of being able to compare two tuple represented versions with a single comparison operator (`<`). For example: `(1, 80) < (1, 60) == True`,  `(11, 0) < (1, 20) == False`
Be mindful tho, that using this technique `(1, 1) < (1, 1) == False` but `(1, 1) < (1, 1, 0) == True)`.

Reply:
`distutils.version` has a version number library that handles version strings (even including letters I think) in a proper order and nice human readability